include/bitcoin/system/math/ec_scalar.hpp

```diff
@@ -52,11 +57,17 @@ class BC_API ec_scalar
     const ec_secret& secret() const;
 
 protected:
-    // These should be const, apart from the need to implement assignment.
-    bool valid_;
-    ec_secret secret_;
+    friend ec_scalar operator+(ec_scalar left, const ec_scalar& right);
+    friend ec_scalar operator-(ec_scalar left, const ec_scalar& right);
+    friend ec_scalar operator*(ec_scalar left, const ec_scalar& right);
+
+    using ec_secret_uniq = std::unique_ptr<ec_secret>;
+
+    ec_secret_uniq secret_;
 };
 
+bool operator==(const ec_scalar& left, const ec_scalar& right);
```

Comment:
Cannot these all be member operators? Better organizationally to have members than free-floating functions, even for operator overrides, where possible. This is a common pattern throughout the libs.

Reply:
I think the common argument is that free floating operators are better than member ones because then LHS can do implicit conversions from other types like `4 == ec_scalar(4)`. Whereas member operators cannot do this conversion so that would be an invalid statement with a member operator.